src/System.Private.Interpreter/src/Internal/Runtime/Interpreter/ILInterpreter.cs

```diff
@@ -2442,5 +2445,189 @@ private void InterpretLoadElement(int token)
+                    break;
+                case TypeFlags.SByte:
+                case TypeFlags.Byte:
+                    callInfo.ReturnValue = StackItem.FromInt32(localVariableSet.GetVar<sbyte>(0));
+                    break;
+                case TypeFlags.Int16:
+                case TypeFlags.UInt16:
+                    callInfo.ReturnValue = StackItem.FromInt32(localVariableSet.GetVar<short>(0));
+                    break;
+                case TypeFlags.Int32:
+                case TypeFlags.UInt32:
+                    callInfo.ReturnValue = StackItem.FromInt32(localVariableSet.GetVar<int>(0));
+                    break;
+                case TypeFlags.Int64:
+                case TypeFlags.UInt64:
+                    callInfo.ReturnValue = StackItem.FromInt64(localVariableSet.GetVar<long>(0));
+                    break;
+                case TypeFlags.IntPtr:
+                case TypeFlags.UIntPtr:
+                    callInfo.ReturnValue = StackItem.FromNativeInt(localVariableSet.GetVar<IntPtr>(0));
+                    break;
+                case TypeFlags.Single:
+                    _stack.Push(StackItem.FromDouble(localVariableSet.GetVar<float>(0)));
+                    break;
+                case TypeFlags.Double:
+                    callInfo.ReturnValue = StackItem.FromDouble(localVariableSet.GetVar<double>(0));
+                    break;
+                case TypeFlags.ValueType:
+                case TypeFlags.Nullable:
+                    callInfo.ReturnValue = StackItem.FromValueType(localVariableSet.GetVar<ValueType>(0));
+                    break;
+                case TypeFlags.Enum:
+                    returnType = returnType.UnderlyingType;
+                    goto getvar;
+                case TypeFlags.Class:
+                case TypeFlags.Interface:
+                case TypeFlags.Array:
+                case TypeFlags.SzArray:
+                    callInfo.ReturnValue = StackItem.FromObjectRef(localVariableSet.GetVar<object>(0));
+                    break;
+                default:
+                    // TODO: Support more complex return types
+                    throw new NotImplementedException();
+            }
+        }
+
+        private void InterpretCall(int token)
+        {
+            MethodDesc method = (MethodDesc)_methodIL.GetObject(token);
+            MethodSignature signature = method.Signature;
+
+            TypeDesc owningType = method.OwningType;
+            TypeDesc returnType = signature.ReturnType;
+
+            if (signature.Length > _stack.Count)
+                ThrowHelper.ThrowInvalidProgramException();
+
+            int delta = (signature.IsStatic ? 1 : 2);
+
+            LocalVariableType[] localVariableTypes = new LocalVariableType[signature.Length + delta];
+            localVariableTypes[0] = new LocalVariableType(returnType.GetRuntimeTypeHandle(), false, returnType.IsByRef);
```

Comment:
Maybe make this throw if any of the parameters is `ByRef`. I think the calling convention converter expects the type to be unwrapped from  `ref Foo` to `Foo` in that case (the RuntimeTypeHandle should be for the non-byref version of the type).